cpp/src/Glacier2/SessionRouterI.cpp

```diff
@@ -143,7 +143,7 @@ class UserPasswordCreateSession final : public CreateSession
         ctx.insert(_context.begin(), _context.end());
         auto self = static_pointer_cast<UserPasswordCreateSession>(shared_from_this());
         _sessionRouter->_verifier->checkPermissionsAsync(_user, _password,
-                                                         [self](bool ok, const string& reason)
+                                                         [self](bool ok, string reason)
                                                          {
                                                              self->checkPermissionsResponse(ok, reason);
```

Comment:
Shouldn't we call `std::move(reason)` and fix `checkPermissionsResponse` to take a string by value too.

Reply:
I'll leave this for a follow-up PR.

Ideally, we should indeed replace most `const string&` parameter by either a `string_view` parameter (meaning the implementation only reads the string) or a `string` parameter (meaning the implementation often adopts the string).

Here, the implementation could adopt the string only in the rare case where we report permission denied exception (definitely not performance critical), so arguably checkResponse's reason should be a string_view, not a const string& or a string.

"could adopt" because the ctor of PermissionDeniedException today has no "adopt string" overload. I guess we could add such overload for consistency with the Slice exception mapping - which would need to be fixed first:

The generated code for an exception holding a string is currently:

```cpp
class E : public ::Ice::UserExceptionHelper<E, ::Ice::UserException>
{
public:

    virtual ~E();

    E(const E&) = default;

    E() = default;

    /**
     * One-shot constructor to initialize all data members.
     */
    E(const ::std::string& data) :
        data(data)
    {
    }
```